pkg/cloud/services/loadbalancer/loadbalancer.go

```diff
@@ -518,12 +518,18 @@ func (s *Service) ReconcileLoadBalancerMember(openStackCluster *infrav1.OpenStac
 	if openStackCluster.Status.APIServerLoadBalancer == nil {
 		return errors.New("network.APIServerLoadBalancer is not yet available in openStackCluster.Status")
 	}
+	if openStackCluster.Spec.ControlPlaneEndpoint == nil || !openStackCluster.Spec.ControlPlaneEndpoint.IsValid() {
+		return errors.New("ControlPlaneEndpoint is not yet set in openStackCluster.Spec")
+	}
 
 	loadBalancerName := getLoadBalancerName(clusterName)
 	s.scope.Logger().Info("Reconciling load balancer member", "loadBalancerName", loadBalancerName)
 
 	lbID := openStackCluster.Status.APIServerLoadBalancer.ID
-	portList := []int{int(openStackCluster.Spec.ControlPlaneEndpoint.Port)}
+	var portList []int
+	if openStackCluster.Spec.ControlPlaneEndpoint != nil {
```

Comment:
Do we really need the if since we already checked for it at L521?
I guess it's ok. Just in case it was a leftover.

Reply:
No, but I've done similar in a few places. In my first pass at this I didn't double-check any pointers, but after hitting a few panics in various iterations I've decided to err on the side of pretty much always checking. Apart from anything else, this is safer against future cut and paste, where somebody might move either use such that one or the other is no longer guarded.

I'd prefer to fix this through the type system, but I'm not having much joy with that today: https://hachyderm.io/@mattb/112077588227331296